modules/assim.sequential/R/sda.enkf.R

```diff
@@ -530,26 +548,66 @@ sda.enkf <- function(settings, obs.mean, obs.cov, IC = NULL, Q = NULL, adjustmen
         }else{
           Cmodel_tobit2space$y.ind <- x.ind
           Cmodel_tobit2space$y.censored <- x.censored
-          Cmodel_tobit2space$muf <- mu.f
-          Cmodel_tobit2space$pf <- solve(Pf)
+          Cmodel_tobit2space$lambda_0 <- diag(10,length(mu.f)) #enkf.params[[t-1]]$Pa
+          Cmodel_tobit2space$mu_0 <- rep(0,length(mu.f)) #enkf.params[[t-1]]$mu.a
           
-          for(i in 1:length(mu.f)) {
+          for(i in seq_along(X)) {
               ## ironically, here we have to "toggle" the value of y.ind[i]
               ## this specifies that when y.ind[i] = 1,
               ## indicator variable is set to 0, which specifies *not* to sample
               valueInCompiledNimbleFunction(Cmcmc_tobit2space$samplerFunctions[[samplerNumberOffset_tobit2space+i]], 'toggle', 1-x.ind[i])
           }
           
         }
-        
+         
         set.seed(0)
-        dat.tobit2space <- runMCMC(Cmcmc_tobit2space, niter = 10000, progressBar=FALSE)
+        dat.tobit2space <- runMCMC(Cmcmc_tobit2space, niter = 50000, progressBar=TRUE)
+        
+        assessParams <- function(dat, Xt, mu_f_TRUE, P_f_TRUE){  
```

Comment:
Is there a reason this function is defined inline within another function like this rather than defining it outside as a stand-alone function as we typically do?